### PR TITLE
Issue1180 Structs as generic arguments are treated like classes

### LINKED ISSUE #1180: Structs as generic arguments are treated like classes
If you use a struct as a generic argument, it will be passed by reference everywhere and the $clone method will not be inserted anywhere.

### Expected

0
5

### Actual

5
5

### Steps To Reproduce

[Live](http://live.bridge.net/#f01f1e291e860821a6ab1b0913c888c7)

```csharp
public struct Vector2
{
    public float x;
    public float y;
}

public class App
{
    [Ready]
    public static void Main()
    {
        List<Vector2> list = new List<Vector2>();
        list.Add(new Vector2() { x = 0.0f, y = 1.0f });
        
        Vector2 vec = list[0];
        
        vec.x = 5.0f;
        
        Window.Alert(list[0].x.ToString()); // Oops.
        Window.Alert(vec.x.ToString());
    }
}
```

### PR DESCRIPTION
**Pull request by vladsch**
Fixes #1180.

### CODE REVIEW

File: Tests/Runner/Batch1/code.js [hunk 1]
```diff
@@ -3103,7 +3103,7 @@ SomeExternalNamespace.SomeNonBridgeClass.prototype.foo = function(){return 1;};
                     x: 5,
                     y: 7
                 } );
-                var c = b.test2(a.$clone());
+                var c = b.test2(a.$clone()).$clone();
```
Comment (LeonidVeriga) [inline Tests/Runner/Batch1/code.js]: @vladsch, this change does not look as expected

File: Tests/Runner/Batch1/code.js [hunk 2]
```diff
@@ -6272,7 +6318,7 @@ SomeExternalNamespace.SomeNonBridgeClass.prototype.foo = function(){return 1;};
                 var a = Bridge.ClientTest.BridgeIssues.Bridge577.someMethodA(1);
                 Bridge.Test.Assert.notNull$1(a, "#577 Bridge577UnitA created");
     
-                var b = Bridge.ClientTest.BridgeIssues.Bridge577.someMethodB(7);
+                var b = Bridge.ClientTest.BridgeIssues.Bridge577.someMethodB(7).$clone();
```
Comment (LeonidVeriga) [inline Tests/Runner/Batch1/code.js]: @vladsch, this change does not look as expected

File: Tests/Runner/Batch1/code.js [hunk 3]
```diff
@@ -8629,7 +8675,7 @@ SomeExternalNamespace.SomeNonBridgeClass.prototype.foo = function(){return 1;};
     
                 var value1 = Bridge.Nullable.getValueOrDefault(test1, 0);
                 var value2 = Bridge.Nullable.getValueOrDefault(test2, new Bridge.ClientTest.BridgeIssues.Bridge762A());
-                var value3 = Bridge.Nullable.getValueOrDefault(test3, new Bridge.ClientTest.BridgeIssues.Bridge762B());
+                var value3 = Bridge.Nullable.getValueOrDefault(test3, new Bridge.ClientTest.BridgeIssues.Bridge762B()).$clone();
```
Comment (LeonidVeriga) [inline Tests/Runner/Batch1/code.js]: @vladsch, this change does not look as expected

File: Tests/Runner/Batch1/code.js (hunk 1 above)
Comment (vladsch) [inline Tests/Runner/Batch1/code.js]: i guess it looks good, we should clone struct when we assign to new variable

File: Tests/Runner/Batch1/code.js (hunk 2 above)
Comment (vladsch) [inline Tests/Runner/Batch1/code.js]: i guess it looks good, we should clone struct when we assign to new variable

**Pull request merged**